core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -128,21 +138,83 @@
   }
 
   /**
+   * An interface extending{@link FieldIndex}, represents a unique index.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface UniqueFieldIndex<T> extends FieldIndex<T> {}
+
+  /**
+   * An interface extending{@link FieldIndex}, represents a nonunique index.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface NonUniqueFieldIndex<T> extends FieldIndex<T> {}
```

Comment:
No, we have two references for the NonUniqueFieldIndex. The indexes for InputStreamAgentSessionId and OutputStreamAgentSessionId are non-unique ones.